File: Array/Consecutive Number Sum.go
package main

import "fmt"

func main() {

	result := consecutiveSum(9)
	fmt.Println(result)
}

func consecutiveSum(n int) int {
	val, ans, i := 2*n, 0, 1

	for i*i <= val {
		if val%i == 0 {
			temp := val / i
			if (temp-i-1)%2 == 0 && (temp-i-1)/2 >= 0 {
				ans++
			}
		}
		i++
	}

	return ans
}

File: Pythons Solutions/Consecutive Number Sum.py
class Solution:
    def numberSum(self, n):
        val, ans, i = 2*n, 0, 1

        while i*i <= n:
            if val % i == 0:
                temp = val/i
                if (temp-i-1)%2 == 0 and (temp-i-1)//2 >=0:
                    ans += 1
            i += 1
        return ans

obj = Solution()
ans = obj.numberSum(9)
print(ans)